substratest/factory.py

```diff
@@ -329,7 +336,7 @@ class _BaseComputePlanSpec(_Spec, abc.ABC):
     aggregatetuples: typing.List[ComputePlanAggregatetupleSpec]
     testtuples: typing.List[ComputePlanTesttupleSpec]
 
-    def add_traintuple(self, algo, dataset, data_samples, in_models=None, tag=''):
+    def add_traintuple(self, algo, dataset, data_samples, in_models=None, tag='', metadata=None):
```

Comment:
These `add_<asset>` are creating compute plan specs. It's been decided to handle compute plan specs in a separate PR, so they should be removed.

Reply:
You're right, I'll do this!